ElasticsearchExporterException: Failed retrieving mappings from index/index templates with pattern

**Describe the bug**

<!-- A clear and concise description of what the bug is.
For existing Camunda Customers,
please also use our [Support Channels](https://camunda.com/services/support/) to
get priority assistance -->

https://console.cloud.google.com/errors/detail/CMjsyJf_4uLtkAE;service=zeebe;time=P7D;locations=global?inv=1&invt=AbjvZQ&project=camunda-saas-int-chaos

**To Reproduce**

<!--
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
- when using the Java client: https://github.com/zeebe-io/zeebe-test-template-java

-->

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->
<details><summary>Full Stacktrace</summary>
 <p>

```
io.camunda.exporter.exceptions.ElasticsearchExporterException: Failed retrieving mappings from index/index templates with pattern [camunda-tenant-8.8.0_*,camunda-web-session-8.8.0_*,operate-decision-8.3.0_*,camunda-authorization-8.8.0_*,tasklist-metric-8.3.0_*,operate-process-8.3.0_*,camunda-user-8.8.0_*,operate-metric-8.3.0_*,tasklist-form-8.4.0_*,camunda-mapping-8.8.0_*,camunda-role-8.8.0_*,operate-decision-requirements-8.3.0_*,tasklist-import-position-8.2.0_*,camunda-group-8.8.0_*,operate-import-position-8.3.0_*]

at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings ( io/camunda.exporter.schema.elasticsearch/ElasticsearchEngineClient.java:171 )
at io.camunda.exporter.schema.SchemaManager.validateIndices ( io/camunda.exporter.schema/SchemaManager.java:300 )
at io.camunda.exporter.schema.SchemaManager.startup ( io/camunda.exporter.schema/SchemaManager.java:63 )
at io.camunda.exporter.CamundaExporter.open ( io/camunda.exporter/CamundaExporter.java:132 )
at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.lambda$openExporter$0 ( io/camunda.zeebe.broker.exporter.stream/ExporterContainer.java:118 )
at io.camunda.zeebe.util.jar.ThreadContextUtil.runCheckedWithClassLoader ( io/camunda.zeebe.util.jar/ThreadContextUtil.java:59 )
at io.camunda.zeebe.util.jar.ThreadContextUtil.runWithClassLoader ( io/camunda.zeebe.util.jar/ThreadContextUtil.java:35 )
at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.openExporter ( io/camunda.zeebe.broker.exporter.stream/ExporterContainer.java:117 )
at io.camunda.zeebe.broker.exporter.stream.ExporterDirector.lambda$startActiveExportingMode$16 ( io/camunda.zeebe.broker.exporter.stream/ExporterDirector.java:529 )
at io.camunda.zeebe.scheduler.retry.BackOffRetryStrategy.run ( io/camunda.zeebe.scheduler.retry/BackOffRetryStrategy.java:51 )
at io.camunda.zeebe.scheduler.ActorJob.invoke ( io/camunda.zeebe.scheduler/ActorJob.java:86 )
at io.camunda.zeebe.scheduler.ActorJob.execute ( io/camunda.zeebe.scheduler/ActorJob.java:43 )
at io.camunda.zeebe.scheduler.ActorTask.execute ( io/camunda.zeebe.scheduler/ActorTask.java:125 )
at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask ( io/camunda.zeebe.scheduler/ActorThread.java:127 )
at io.camunda.zeebe.scheduler.ActorThread.doWork ( io/camunda.zeebe.scheduler/ActorThread.java:104 )
at io.camunda.zeebe.scheduler.ActorThread.run ( io/camunda.zeebe.scheduler/ActorThread.java:224 )
Caused by: java.io.IOException

…
at org.elasticsearch.client.RestClient.extractAndWrapCause ( RestClient.java:939 )
at org.elasticsearch.client.RestClient.performRequest ( RestClient.java:304 )
at org.elasticsearch.client.RestClient.performRequest ( RestClient.java:292 )
at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest ( RestClientHttpClient.java:92 )
at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest ( ElasticsearchTransportBase.java:145 )
at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping ( ElasticsearchIndicesClient.java:1257 )
at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping ( ElasticsearchIndicesClient.java:1275 )
at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings ( ElasticsearchEngineClient.java:329 )
at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings ( ElasticsearchEngineClient.java:154 )
…
Caused by: java.net.UnknownHostException

…
at org.elasticsearch.client.RestClient.performRequest ( RestClient.java:300 )
at org.elasticsearch.client.RestClient.performRequest ( RestClient.java:292 )
at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest ( RestClientHttpClient.java:92 )
at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest ( ElasticsearchTransportBase.java:145 )
at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping ( ElasticsearchIndicesClient.java:1257 )
at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping ( ElasticsearchIndicesClient.java:1275 )
at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings ( ElasticsearchEngineClient.java:329 )
at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings ( ElasticsearchEngineClient.java:154 )
…
```
</p>
</details>

**Environment:**
- OS: <!-- [e.g. Linux] -->
- Zeebe Version: <!-- [e.g. 0.20.0] -->
- Configuration: <!-- [e.g. exporters etc.] -->

@ChrisKujawa / @korthout Does anybody has more context to this? A customer trying alpha3-RC4 at the moment runs into the following stack trace - and Schema creation doesn't work. Is this the same issue? Is this a bug? 

```
      io.camunda.exporter.schema.SchemaManager - Schema creation is enabled. Start Schema management.
Exception in thread "main" io.camunda.exporter.exceptions.ElasticsearchExporterException: Failed retrieving mappings from index/index templates with pattern [txxxxx-operate-metric-8.3.0_*,txxxxx-camunda-user-8.7.0_*,txxxxx-operate-decision-8.3.0_*,txxxxx-camunda-group-8.7.0_*,txxxxx-tasklist-form-8.4.0_*,txxxxx-camunda-role-8.7.0_*,txxxxx-camunda-mapping-8.7.0_*,txxxxx-operate-import-position-8.3.0_*,txxxxx-tasklist-metric-8.3.0_*,txxxxx-camunda-tenant-8.7.0_*,txxxxx-operate-process-8.3.0_*,txxxxx-tasklist-import-position-8.2.0_*,      io.camunda.exporter.schema.SchemaManager - Schema creation is enabled. Start Schema management.

Exception in thread "main" io.camunda.exporter.exceptions.ElasticsearchExporterException: Failed retrieving mappings from index/index templates with pattern [txxxxx-operate-metric-8.3.0_*,txxxxx-camunda-user-8.7.0_*,txxxxx-operate-decision-8.3.0_*,txxxxx-camunda-group-8.7.0_*,txxxxx-tasklist-form-8.4.0_*,txxxxx-camunda-role-8.7.0_*,txxxxx-camunda-mapping-8.7.0_*,txxxxx-operate-import-position-8.3.0_*,txxxxx-tasklist-metric-8.3.0_*,txxxxx-camunda-tenant-8.7.0_*,txxxxx-operate-process-8.3.0_*,txxxxx-tasklist-import-position-8.2.0_*,txxxxx-operate-decision-requirements-8.3.0_*,txxxxx-camunda-authorization-8.7.0_*,txxxxx-camunda-web-session-8.7.0_*]
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:164)
        at io.camunda.exporter.schema.SchemaManager.validateIndices(SchemaManager.java:180)
        at io.camunda.exporter.schema.SchemaManager.startup(SchemaManager.java:47)
        at io.camunda.application.StandaloneSchemaManager.main(StandaloneSchemaManager.java:103)
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:919)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:91)
        at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:144)
        at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1210)
        at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1226)
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings(ElasticsearchEngineClient.java:272)
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:147)
        ... 3 more
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)
        at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:506)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
        at java.base/java.lang.Thread.run(Thread.java:1583)-operate-decision-requirements-8.3.0_*,txxxxx-camunda-authorization-8.7.0_*,txxxxx-camunda-web-session-8.7.0_*]
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:164)
        at io.camunda.exporter.schema.SchemaManager.validateIndices(SchemaManager.java:180)
        at io.camunda.exporter.schema.SchemaManager.startup(SchemaManager.java:47)
        at io.camunda.application.StandaloneSchemaManager.main(StandaloneSchemaManager.java:103)
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:919)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:91)
        at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:144)
        at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1210)
        at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1226)
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings(ElasticsearchEngineClient.java:272)
        at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:147)
        ... 3 more
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
       at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)
        at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:506)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
       at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
        at java.base/java.lang.Thread.run(Thread.java:1583)
```

Happened again in `8.8.0-alpha4` [GCP log](https://console.cloud.google.com/logs/query;cursorTimestamp=2025-05-13T11:48:31.169410275Z;endTime=2025-05-13T12:19:00.695Z;pinnedLogId=2025-05-13T11:48:30.695375843Z%2Fpf1zvc9ql65lwoe9;query=logName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22production-worker-1%22%0Aresource.labels.namespace_name%3D%22c16204ec-c864-4e2a-bb4c-c1eefe2a0fc9-zeebe%22%0Aresource.labels.location%3D%22europe-west1%22%0Aresource.labels.project_id%3D%22camunda-cloud-240911%22%0Aresource.labels.pod_name%3D%22zeebe-1%22%0Aresource.labels.container_name%3D%22zeebe%22%0Atimestamp%3D%222025-05-13T11:48:31.169410275Z%22%0AinsertId%3D%224mh5fod7ayfszt88%22;startTime=2025-05-13T11:19:00.695Z?project=camunda-cloud-240911)

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->
<details><summary>Full Stacktrace</summary>
 <p>

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'searchEngineSchemaInitializer' defined in class path resource [io/camunda/application/commons/search/SearchEngineDatabaseConfiguration.class]: Failed retrieving mappings from index/index templates with pattern [camunda-group-8.8.0_*,tasklist-user-1.4.0_*,tasklist-form-8.4.0_*,camunda-user-8.8.0_*,camunda-tenant-8.8.0_*,operate-process-8.3.0_*,tasklist-metric-8.3.0_*,operate-decision-requirements-8.3.0_*,camunda-authorization-8.8.0_*,operate-user-1.2.0_*,camunda-mapping-8.8.0_*,camunda-role-8.8.0_*,operate-metric-8.3.0_*,tasklist-import-position-8.2.0_*,operate-decision-8.3.0_*,camunda-web-session-8.8.0_*,operate-import-position-8.3.0_*]: Connection refused
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1812) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:346) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1155) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1121) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1056) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987) ~[spring-context-6.2.3.jar:6.2.3]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627) ~[spring-context-6.2.3.jar:6.2.3]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.4.3.jar:3.4.3]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752) [spring-boot-3.4.3.jar:3.4.3]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) [spring-boot-3.4.3.jar:3.4.3]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) [spring-boot-3.4.3.jar:3.4.3]
	at io.camunda.application.StandaloneBroker.main(StandaloneBroker.java:30) [camunda-zeebe-8.8.0-alpha4.jar:8.8.0-alpha4]
Caused by: io.camunda.search.schema.exceptions.SearchEngineException: Failed retrieving mappings from index/index templates with pattern [camunda-group-8.8.0_*,tasklist-user-1.4.0_*,tasklist-form-8.4.0_*,camunda-user-8.8.0_*,camunda-tenant-8.8.0_*,operate-process-8.3.0_*,tasklist-metric-8.3.0_*,operate-decision-requirements-8.3.0_*,camunda-authorization-8.8.0_*,operate-user-1.2.0_*,camunda-mapping-8.8.0_*,camunda-role-8.8.0_*,operate-metric-8.3.0_*,tasklist-import-position-8.2.0_*,operate-decision-8.3.0_*,camunda-web-session-8.8.0_*,operate-import-position-8.3.0_*]: Connection refused
	at io.camunda.search.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:179) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.validateIndices(SchemaManager.java:327) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.initializeSchema(SchemaManager.java:71) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.github.resilience4j.retry.Retry.lambda$decorateRunnable$7(Retry.java:375) ~[resilience4j-retry-2.3.0.jar:2.3.0]
	at io.camunda.zeebe.util.retry.RetryDecorator.decorate(RetryDecorator.java:71) ~[zeebe-util-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.startup(SchemaManager.java:65) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.application.commons.search.SearchEngineSchemaInitializer.afterPropertiesSet(SearchEngineSchemaInitializer.java:42) ~[camunda-zeebe-8.8.0-alpha4.jar:8.8.0-alpha4]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1859) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1808) ~[spring-beans-6.2.3.jar:6.2.3]
	... 16 more
Caused by: java.net.ConnectException: Connection refused
	at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:934) ~[elasticsearch-rest-client-8.16.4.jar:8.16.4]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304) ~[elasticsearch-rest-client-8.16.4.jar:8.16.4]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292) ~[elasticsearch-rest-client-8.16.4.jar:8.16.4]
	at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:92) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:145) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1257) ~[elasticsearch-java-8.16.4.jar:?]
	at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1275) ~[elasticsearch-java-8.16.4.jar:?]
	at io.camunda.search.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings(ElasticsearchEngineClient.java:364) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:162) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.validateIndices(SchemaManager.java:327) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.initializeSchema(SchemaManager.java:71) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.github.resilience4j.retry.Retry.lambda$decorateRunnable$7(Retry.java:375) ~[resilience4j-retry-2.3.0.jar:2.3.0]
	at io.camunda.zeebe.util.retry.RetryDecorator.decorate(RetryDecorator.java:71) ~[zeebe-util-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.search.schema.SchemaManager.startup(SchemaManager.java:65) ~[camunda-schema-manager-8.8.0-alpha4.jar:8.8.0-alpha4]
	at io.camunda.application.commons.search.SearchEngineSchemaInitializer.afterPropertiesSet(SearchEngineSchemaInitializer.java:42) ~[camunda-zeebe-8.8.0-alpha4.jar:8.8.0-alpha4]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1859) ~[spring-beans-6.2.3.jar:6.2.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1808) ~[spring-beans-6.2.3.jar:6.2.3]
	... 16 more
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at java.base/sun.nio.ch.Net.pollConnectNow(Unknown Source) ~[?:?]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(Unknown Source) ~[?:?]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvent(DefaultConnectingIOReactor.java:174) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvents(DefaultConnectingIOReactor.java:148) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.execute(AbstractMultiworkerIOReactor.java:351) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager.execute(PoolingNHttpClientConnectionManager.java:221) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.client.CloseableHttpAsyncClientBase$1.run(CloseableHttpAsyncClientBase.java:64) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at java.base/java.lang.Thread.run(Unknown Source) ~[?:?]
```
</p>
</details>


FYI @ChrisKujawa

--- a/schema-manager/src/main/java/io/camunda/search/schema/elasticsearch/ElasticsearchEngineClient.java
+++ b/schema-manager/src/main/java/io/camunda/search/schema/elasticsearch/ElasticsearchEngineClient.java
@@ -43,6 +43,7 @@ import io.camunda.search.schema.config.IndexConfiguration;
 import io.camunda.search.schema.exceptions.IndexSchemaValidationException;
 import io.camunda.search.schema.exceptions.SearchEngineException;
 import io.camunda.search.schema.utils.SearchEngineClientUtils;
+import io.camunda.search.schema.utils.SuppressLogger;
 import io.camunda.webapps.schema.descriptors.IndexDescriptor;
 import io.camunda.webapps.schema.descriptors.IndexTemplateDescriptor;
 import io.camunda.webapps.schema.descriptors.index.ImportPositionIndex;
@@ -55,11 +56,11 @@ import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
 import java.util.stream.Collectors;
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ElasticsearchEngineClient implements SearchEngineClient {
-  private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchEngineClient.class);
+  private static final SuppressLogger LOG =
+      new SuppressLogger(LoggerFactory.getLogger(ElasticsearchEngineClient.class));
   private static final Slices AUTO_SLICES =
       Slices.of(slices -> slices.computed(SlicesCalculation.Auto));
   private final ElasticsearchClient client;

--- a/schema-manager/src/main/java/io/camunda/search/schema/opensearch/OpensearchEngineClient.java
+++ b/schema-manager/src/main/java/io/camunda/search/schema/opensearch/OpensearchEngineClient.java
@@ -20,6 +20,7 @@ import io.camunda.search.schema.config.IndexConfiguration;
 import io.camunda.search.schema.exceptions.IndexSchemaValidationException;
 import io.camunda.search.schema.exceptions.SearchEngineException;
 import io.camunda.search.schema.utils.SearchEngineClientUtils;
+import io.camunda.search.schema.utils.SuppressLogger;
 import io.camunda.webapps.schema.descriptors.IndexDescriptor;
 import io.camunda.webapps.schema.descriptors.IndexTemplateDescriptor;
 import io.camunda.webapps.schema.descriptors.index.ImportPositionIndex;
@@ -56,11 +57,11 @@ import org.opensearch.client.opensearch.indices.PutMappingRequest;
 import org.opensearch.client.opensearch.indices.get_index_template.IndexTemplateItem;
 import org.opensearch.client.opensearch.indices.get_index_template.IndexTemplateSummary;
 import org.opensearch.client.opensearch.indices.put_index_template.IndexTemplateMapping;
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class OpensearchEngineClient implements SearchEngineClient {
-  private static final Logger LOG = LoggerFactory.getLogger(OpensearchEngineClient.class);
+  private static final SuppressLogger LOG =
+      new SuppressLogger(LoggerFactory.getLogger(OpensearchEngineClient.class));
   private static final String OPERATE_DELETE_ARCHIVED_POLICY =
       "/schema/opensearch/create/policy/operate_delete_archived_indices.json";
   private static final long AUTO_SLICES = 0; // see OS docs; 0 means auto

--- a/schema-manager/src/main/java/io/camunda/search/schema/utils/SuppressLogger.java
+++ b/schema-manager/src/main/java/io/camunda/search/schema/utils/SuppressLogger.java
@@ -1,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.schema.utils;
+
+import java.net.ConnectException;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.util.Set;
+import org.slf4j.Logger;
+
+// Downgrade transient connection issues to ElasticSearch/OpenSearch to a WARN
+public class SuppressLogger {
+  private static final Set<Class<? extends Exception>> SUPPRESSED_EXCEPTIONS =
+      Set.of(SocketTimeoutException.class, ConnectException.class, SocketException.class);
+  private final Logger logger;
+
+  public SuppressLogger(final Logger logger) {
+    this.logger = logger;
+  }
+
+  public void error(final String message, final Throwable t) {
+    if (SUPPRESSED_EXCEPTIONS.contains(t.getClass())) {
+      logger.warn("[downgraded from ERROR]: {} ", message, t);
+    } else {
+      logger.error(message, t);
+    }
+  }
+
+  public void debug(final String message, final Object... args) {
+    logger.debug(message, args);
+  }
+
+  public void warn(final String message, final Object... args) {
+    logger.warn(message, args);
+  }
+}

